### DISCUSSION TITLE
Support Amazon DSP API

### DISCUSSION THREAD

**Discussion by twistedFantasy** [denisneuf/python-amazon-ad-api#52] [Ideas]
First of all, great package, just played with it a little bit and everything works without any problems, thank you for your good work here! Any plans to support [Amazon DSP API](https://advertising.amazon.com/API/docs/en-us/dsp/overview) functionality in this package? 
Even though most of the Amazon DSP API is still in beta(see attached image below), but at least [Reports API](https://advertising.amazon.com/API/docs/en-us/dsp-reports-beta-3p/#/Reports) is already stable and can be used in the production systems. Or maybe you faced some unexpected issue during implementation of it? Will be happy to help you with the implementation of this Amazon DSP API(Reports) in the scope of this package if you don't have plans to implement it by yourself, or don't have time to do it. Just give me your thoughts about it here.

![Screenshot from 2022-06-14 10-55-05](https://user-images.githubusercontent.com/37516294/173562130-edc98ed1-04f9-4a8c-8385-33612afe24f4.png)

**Comment by denisneuf** [comment]
Hi, the main problem was not the beta status of the endpoints, others has been implemented in that status.
The problem is that I don't have in my hands any account to test the endpoints so I decide wait until get some kind of access to test it.
It could be implement blindly but Amazon api's are often tricky and a little bit unpredictable. The other endpoints has been tested against sandbox (the ones that admit that feature) and production but this DSP I have no way to test.
So I will be happy if you could help with that I really appreciate if you could contribute to include it in the package.
So if you got familiar with the organization on the package you could submit any pull request to make this package more complete.

**Reply by twistedFantasy** [reply]
Thank you for a quick response! I agree that Amazon api's are often tricky and a little bit unpredictable. Yes, I have access to the Amazon account with DSP so I can implement and test it in the scope of your package. Currently I am under investigation phase, but very soon move to the implementation basic Amazon Ads reports automation(Sponsored Products, Sponsored Display, and Sponsored Brands) with the help of your awesome package and after this will move to the DSP reports and provide you a pull request. I already looked through the source code of your package, everything is clear and good organized, so shouldn't be a very big changes to add this DSP reports support. Thanks again for your work and time!
